lib/App/DuckPAN/Web.pm

```diff
@@ -412,4 +415,80 @@ sub request {
 	return $response;
 }
 
+
+#inject some mock results into the SERP to make it look a little more real
+sub inject_mock_content {
+
+    my $root= shift;
+
+    #inject a mock ad into the page
+    my $ad_container = $root->look_down(id => "ads");
+    $ad_container->attr("style", "display: block");
+
+    my $ad = HTML::TreeBuilder->new_from_content(
+        q(<div id="ra-0" class="result results_links highlight_a  result--ad  highlight_sponsored  sponsored highlight highlight_sponsored_hover" data-nir="1">
+            <div class="result__body links_main links_deep">
+                <a href="https://duck.co/help/company/advertising-and-affiliates" class="result__badge  badge--ad">Ad</a>
+                <h2 class="result__title">
+                <a class="result__a" href="">Lorem ipsum Culpa ex adipisicing.</a>
+                <a class="result__check" href="">
+                    <span class="result__check__tt">Lorem ipsum Consectetur nostrud id quis in ut.</span>
+                </a>
+                </h2>
+                <div class="result__snippet">
+                    <a href="">Lorem ipsum Nisi aute velit sit dolore sit amet fugiat consequat aute reprehenderit in dolore deserunt.</a>
+                </div>
+                <div class="result__extras">
+                    <div class="result__extras__url">
+                        <a class="result__url" href="#">duckduckgo.com</a>
+                    </div>
+                </div>
+            </div>
+        </div>
+        )
+    )->guts;
+
+    $ad_container->push_content( $ad );
+
+    #inject some mock ad into the page
+    my $links_container = $root->look_down(id => "links");
+
+    for (1..4){
+        $links_container->push_content(
+            HTML::TreeBuilder->new_from_content(
```

Comment:
Yes and no. DuckPAN makes a request to duckduckgo.com with a special query: ["duckduckhack-template-for-spice2"](https://duckduckgo.com/?q=duckduckhack-template-for-spice2). The DDG backend sees this and in that case doesn't bother putting in any results. Instead it puts in some placeholder text that DuckPAN uses to inject the `<script>` tags into.

So if they `id` for the ads and results divs, changes, this would break. But the result would just be the fake ads would be gone. I don't suspect the HTML is going to change any time soon though ;)